### PR TITLE
[Chopbox] [#108003186] Edit and delete comments

### PR DESCRIPTION
**Pull request by vdugeri**

### CODE REVIEW

File: app/Http/routes.php [hunk 1]
```diff
@@ -83,6 +83,25 @@
 Route::post('comment', 'CommentController@addComment');
 
 
+Route::post('comment/delete/{id}',[
```
Comment (vdugeri) [inline app/Http/routes.php]: Issue resolved

File: app/Providers/AuthServiceProvider.php [hunk 2]
```diff
@@ -36,5 +36,11 @@ public function boot(GateContract $gate)
            $userToEdit  = User::find($id);
             return $user->id == $userToEdit->id;
         });
+
+
+        $gate->define('edit-comment', function($user, $comment){
```
Comment (vdugeri) [inline app/Providers/AuthServiceProvider.php]: Issue fixed

File: app/Providers/AuthServiceProvider.php [hunk 3]
```diff
@@ -36,5 +36,11 @@ public function boot(GateContract $gate)
...
+           return $user->id = $comment->user->id;
```
Comment (vdugeri) [inline app/Providers/AuthServiceProvider.php]: Issue fixed

File: public/js/jquery.toaster.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+		{
+			var $toaster = this.gettoaster();
+			var $toast  = $(settings.toast.template.replace('%priority%', priority)).hide().css(settings.toast.css).addClass(settings.toast['class']);
+
+			$('.title', $toast).css(settings.toast.csst).html(title);
+			$('.message', $toast).css(settings.toast.cssm).html(message);
+
+			if ((settings.debug) && (window.console))
+			{
+				console.log(toast);
```
Comment (vdugeri) [inline public/js/jquery.toaster.js:63]: This is a plugin. I have no business editing this code.

File: public/js/deleteComment.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        findComment(id, function(comment){
+            var text = comment.comment;
+            BootstrapDialog.show({
+                title : 'Edit Comment',
+                message: function(dialog){
+                    var content = $("<div class='form-group'>" +
+                    "<input type='text' class='form-control' name='comment' value='"+text+"'>" +
+                    "</div>");
+
+                    var footerButton = dialog.getButton('edit-button');
```
Comment (vdugeri) [inline public/js/deleteComment.js]: unnecessary declaration removed

File: public/js/deleteComment.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                },
+                {
+                    label: "Yep",
+                    cssClass: 'btn-primary',
+                    action: function(dialog){
+                        $.ajax({
+                            url: 'comment/delete/'+id,
+                            type: 'post',
+                            data: {id: id},
+                            success: function(response){
```
Comment (vdugeri) [inline public/js/deleteComment.js]: unused parameter removed

File: public/js/deleteComment.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                                dialog.close();
+
+                                $.toaster({
+                                    priority : 'success',
+                                    title : 'Sucess',
+                                    message : 'comment deleted'
+                                })
+
+                            },
+                            error: function(response){
```
Comment (vdugeri) [inline public/js/deleteComment.js]: Unused parameter removed

File: public/js/deleteComment.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                }
+            }]
+        });
+    }
+
+    function editComment(id)
+    {
+        findComment(id, function(comment){
+            var text = comment.comment;
+            BootstrapDialog.show({
```
Comment (vdugeri) [inline public/js/deleteComment.js:69]: Lol. look again.

File: public/js/deleteComment.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    $('.edit-comment').on('click', function(e){
+        e.preventDefault();
+        var id = $(this).data('id');
+        editComment(id);
+    });
+
+
+    function deleteComment(id)
+    {
+        BootstrapDialog.show({
```
Comment (vdugeri) [inline public/js/deleteComment.js:21]: You need glasses

File: public/js/deleteComment.js [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        $.ajax({
+            url: 'comment/edit/'+id,
+            type: 'post',
+            data: {comment: comment},
+            success: function(response){
+                $('comment#comment-body-'+id).empty().append(response.comment.comment);
+                dialog.close();
+            }
+        })
```
Comment (vdugeri) [inline public/js/deleteComment.js]: Fixed

File: app/Providers/AuthServiceProvider.php [hunk 11]
```diff
@@ -32,9 +32,15 @@ public function boot(GateContract $gate)
...
 
-        $gate->define('edit-profile', function($user, $id){
-           $userToEdit  = User::find($id);
+        $gate->define('edit-profile', function($user, $id) {
+            $userToEdit  = User::find($id);
             return $user->id == $userToEdit->id;
         });
+
+
+        $gate->define('edit-comment', function($user, $comment) {
```
Comment (vdugeri) [inline app/Providers/AuthServiceProvider.php:41]: Lol. Let me educate you. Your expectations are inaccurate.

File: app/Providers/AuthServiceProvider.php [hunk 12]
```diff
@@ -32,9 +32,15 @@ public function boot(GateContract $gate)
...
+           return $user->id = $comment->user->id;
```
Comment (vdugeri) [inline app/Providers/AuthServiceProvider.php]: Indentation fixed

File: public/js/deleteComment.js [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    function updateComment(comment, id, dialog)
+    {
+        $.ajax({
+            url: 'comment/edit/'+id,
+            type: 'post',
+            data: {comment: comment},
+            success: function(response){
+                $('comment#comment-body-'+id).empty().append(response.comment.comment);
```
Comment (vdugeri) [inline public/js/deleteComment.js]: lol. Ok

File: public/js/deleteComment.js [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+                        label: "Done",
+                        cssClass: 'btn-primary',
+                        id: 'edit-button',
+                        action: function(dialog){
+                            var comment = dialog.getModalBody().find('input').val();
+                            updateComment(comment, id, dialog);
+                        }
+                    }
+                ]
+            })
```
Comment (vdugeri) [inline public/js/deleteComment.js]: fixed

File: public/js/deleteComment.js [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+                        action: function(dialog){
+                            dialog.close();
+                        }
+                    },
+                    {
+                        label: "Done",
+                        cssClass: 'btn-primary',
+                        id: 'edit-button',
+                        action: function(dialog){
+                            var comment = dialog.getModalBody().find('input').val();
```
Comment (vdugeri) [inline public/js/deleteComment.js]: fixed

File: public/js/deleteComment.js [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+                buttons: [
+                    {
+                        label: 'Cancel',
+                        cssClass: 'btn-primary',
+                        action: function(dialog){
+                            dialog.close();
+                        }
+                    },
+                    {
+                        label: "Done",
```
Comment (andela-vdugeri) [inline public/js/deleteComment.js]: Show me a better way to do it. If you know.

Comment (vdugeri) [inline public/js/deleteComment.js]: fixed

**Pull request merged**